### PR TITLE
chore(pkger): refactor parser notification endpoint state out into stateful type

### PR DESCRIPTION
**Pull request by jsteenb2**
references: #17434

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)

### CODE REVIEW

File: kit/transport/http/api.go [hunk 1]
```diff
@@ -165,20 +165,35 @@ func (a *API) Respond(w http.ResponseWriter, status int, v interface{}) {
 	}
 
 	w.Header().Set("Content-Type", "application/json; charset=utf-8")
-	enc := json.NewEncoder(writer)
+
+	// this marshal block is to catch failures before they hit the http writer.
```
Comment (jsteenb2) [inline kit/transport/http/api.go:169]: found this issue while testing. I had set an 0 value influxdb.ID type and it exploded in my face during encoding. However, my tests never caught it b/c a 200 was written during encoding. This fixes that up and also makes it viable for anyone else that may have hit this issue.

File: pkger/models.go [hunk 2]
```diff
@@ -877,309 +856,6 @@ type SummaryVariable struct {
...
-			Field: fieldMin,
-			Msg:   "min must be < max",
-		})
-	}
-	return vErrs
-}
-
-func toInfluxThresholds(thresholds ...threshold) []icheck.ThresholdConfig {
-	var iThresh []icheck.ThresholdConfig
-	for _, th := range thresholds {
```
Comment (jsteenb2) [inline pkger/models.go:929]: vast majority of LOCs coming from moving these bits around to their new homes

File: kit/transport/http/api.go (hunk 1 above)
Comment (bthesorceror) [reply kit/transport/http/api.go:169]: did you mean 'body' instead of 'bod' here?

Comment (jsteenb2) [reply kit/transport/http/api.go:169]: good catch 👍, I did mean body

**Pull request merged**